packages/mds-compliance-service/repository/index.ts

```diff
@@ -0,0 +1,175 @@
+import { InsertReturning, RepositoryError, ReadWriteRepository } from '@mds-core/mds-repository'
+import { isDefined, NotFoundError, now } from '@mds-core/mds-utils'
+import { UUID } from '@mds-core/mds-types'
+import {
+  ComplianceSnapshotDomainModel,
+  GetComplianceSnapshotsByTimeIntervalOptions,
+  GetComplianceSnapshotOptions
+} from '../@types'
+import { ComplianceSnapshotEntityToDomain, ComplianceSnapshotDomainToEntityCreate } from './mappers'
+import { ComplianceSnapshotEntity, ComplianceSnapshotEntityModel } from './entities/compliance-snapshot-entity'
+import migrations from './migrations'
+
+class ComplianceSnapshotReadWriteRepository extends ReadWriteRepository {
+  public getComplianceSnapshot = async ({
+    compliance_snapshot_id,
+    provider_id,
+    policy_id,
+    compliance_as_of = now()
+  }: GetComplianceSnapshotOptions): Promise<ComplianceSnapshotDomainModel> => {
+    const { connect } = this
+    try {
+      const connection = await connect('ro')
+      if (isDefined(compliance_snapshot_id)) {
+        const entity = await connection.getRepository(ComplianceSnapshotEntity).findOne({
+          where: {
+            compliance_snapshot_id
+          }
+        })
+        if (!entity) {
+          throw new NotFoundError(`ComplianceSnapshot ${compliance_snapshot_id} not found`)
+        }
+        return ComplianceSnapshotEntityToDomain.map(entity)
+      }
```

Comment:
Is there any particular reason we have two separate DB calls here, one that handles if the user provides a `compliance_snapshot_id` option, and one that handles all of the other options? 
I imagine it may make sense to _only_ look up by the `compliance_snapshot_id` if there is one, but I think for clarity that probably belongs in a separate function, `getComplianceSnapshotById` or something. As a user, I would assume that unless the typing is exclusive (e.g. `{ compliance_snapshot_id: UUID } | Partial<{ provider_id: UUID, policy_id: UUID, compliance_as_of: UUID }>`), all of the options have the same _weight_ to them, and if I provided all of them it would result in the combination of all of them, not potentially basing the result on only one of them and ignoring the others.

Reply:
Cause this is how the options for the API endpoint to fetch a single snapshot work.